modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/CSPComplianceCheck.java

```diff
@@ -162,24 +181,70 @@ private void _checkIllegalTags(
 
 				int lineNumber = getLineNumber(content, x);
 
+				if (fileName.endsWith(".ftl") &&
+					_illegalTagAuiReplacements.contains(tagName)) {
+
+					if (_illegalTagAuiReplacements.contains(tagName)) {
+						_checkMissingAttribute(
+							fileName, tagName, "${nonceAttribute}", tagString,
+							lineNumber);
+					}
+				}
+				else if (fileName.endsWith(".vm") &&
+						 _illegalTagAuiReplacements.contains(tagName)) {
+
+					_checkMissingAttribute(
+						fileName, tagName, "$nonceAttribute", tagString,
+						lineNumber);
+				}
+			}
+
+			x = -1;
+
+			while (true) {
+				x = lowerCaseContent.indexOf("</" + tagName, x + 1);
+
+				if (x == -1) {
+					break;
+				}
+
+				String tagString = getEndingTag(content, x);
+
+				if (Validator.isNull(tagString)) {
+					continue;
+				}
+
+				int lineNumber = getLineNumber(content, x);
+
 				if (fileName.endsWith(".jsp") || fileName.endsWith(".jspf") ||
 					fileName.endsWith(".jspx")) {
 
+					if (_illegalTagAuiReplacements.contains(tagName)) {
+						addMessage(
+							fileName,
+							StringBundler.concat(
+								"Use <aui:", tagName, "> tag instead of <",
+								tagName, ">, see LPD-18227"),
+							lineNumber);
+					}
+					else {
+						addMessage(
+							fileName,
+							StringBundler.concat(
+								"Remove usage of </", tagName,
+								"> tag, see LPD-47204"),
+							lineNumber);
+					}
+				}
+
+				if ((fileName.endsWith(".ftl") || fileName.endsWith(".vm")) &&
+					!_illegalTagAuiReplacements.contains(tagName)) {
+
 					addMessage(
 						fileName,
 						StringBundler.concat(
-							"Use <aui:", tagName, "> tag instead of <", tagName,
-							">, see LPD-18227"),
-						lineNumber);
-				}
-				else if (fileName.endsWith(".ftl")) {
-					_checkMissingAttribute(
-						fileName, tagName, "${nonceAttribute}", tagString,
-						lineNumber);
-				}
-				else if (fileName.endsWith(".vm")) {
-					_checkMissingAttribute(
-						fileName, tagName, "$nonceAttribute", tagString,
+							"Remove usage of </", tagName,
+							"> tag, see LPD-47204"),
```

Comment:
We can simplify and collapse the logic here a bit. Something like this: 
```java
int lineNumber = getLineNumber(content, x);

if (_illegalTagAuiReplacements.contains(tagName)) {
	if (fileName.endsWith(".jsp") ||
		fileName.endsWith(".jspf") || fileName.endsWith(".jspx")) {

		addMessage(
			fileName,
			StringBundler.concat(
				"Use <aui:", tagName, "> tag instead of <",
				tagName, ">, see LPD-18227"),
			lineNumber);
	}

	continue;
}

addMessage(
	fileName, 
	StringBundler.concat(
		"Remove usage of </", tagName, "> tag, see LPD-47204"), 
	lineNumber);
```